cfme/tests/cloud/test_quota.py

```diff
@@ -298,3 +305,192 @@ def test_project_quota_enforce_via_lifecycle_cloud(
+    dialog_rest.action.delete()
+
+
+def get_result(request, appliance, catalog, catalog_item_name, dialog_values=None):
+    """Returns the quota requested by particular type of flavor type"""
+    service_catalogs = ServiceCatalogs(appliance, catalog, catalog_item_name, dialog_values)
+    service_catalogs.order()
+    # nav to requests page to check quota validation
+    request_description = 'Provisioning Service [{0}] from [{0}]'.format(catalog_item_name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui')
+    request.addfinalizer(provision_request.remove_request)
+    assert provision_request.row.reason.text == "Quota Exceeded"
+    last_message = provision_request.row.last_message.text
+    result = re.findall(r'requested.*\w', last_message)
+
+    # Service request needs to delete because we are not able to order same catalog item multiple
+    # times using automation.
+    delete_request = appliance.rest_api.collections.service_requests.get(
+        description=request_description)
+    delete_request.action.delete()
+
+    return result
+
+
+# first arg of parametrize is the list of fixture or parameter,
+# second arg is a list of lists, with a test is to be generated
+# indirect is the list where we define which fixture is to be passed values indirectly.
+@pytest.mark.meta(blockers=[BZ(1704439)])
+@pytest.mark.tier(1)
+@pytest.mark.parametrize(
+    ['set_roottenant_quota'],
+    [
+        [('storage', 0.001)]
+    ],
+    indirect=['set_roottenant_quota'],
+    ids=['max_storage']
+)
+def test_custom_service_dialog_quota_flavors(request, provider, provisioning, dialog, catalog,
+                                             appliance, automate_flavour_method, admin_email,
+                                             set_roottenant_quota):
+    """Test quota with instance/flavor type in custom dialog
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/4h
+        startsin: 5.8
+        casecomponent: Quota
+
+    Bugzilla:
+        1499193
+        1581288
+        1657628
+    """
+    catalog_item = create_catalog_item(appliance, provider, provisioning, dialog, catalog)
+    request.addfinalizer(catalog_item.delete_if_exists)
+    result = []
+
+    # Fetching all the flavours related to particular provider and collecting two flavors randomly
+    flavors = random.sample(appliance.rest_api.collections.flavors.all, 2)
+
+    # Ordering service catalog item with different flavor types
+    for flavor in flavors:
+        flavor_type = {'option_0_instance_type': flavor.name}
+        requested_storage = get_result(
+            request=request, appliance=appliance, catalog=catalog_item.catalog,
+            catalog_item_name=catalog_item.name, dialog_values=flavor_type
+        )
+        result.append(requested_storage)
+    assert result[0] != result[1]
```

Comment:
Add an inline comment about what it is trying to assert here 
requested storage of different flavors should not match ? Is there any case where it can match ?

Reply:
yes. It can match if we order same cataog with same ```flavor``` type via dynamic dialog.

Reply:
:+1: